File: src/main/java/com/github/sarhatabaot/kraken/core/ticks/TickUtil.java
package com.github.sarhatabaot.kraken.core.ticks;

/**
 * @author sarhatabaot
 */
public class TickUtil {
    private TickUtil() {
        throw new UnsupportedOperationException();
    }

    public static long secondsToTicks(int seconds) {
        return 20L * seconds;
    }

    public static long minutesToTicks(int minutes) {
        return secondsToTicks(60 * minutes);
    }

    public static long hoursToTicks(int hours) {
        return minutesToTicks(60 * hours);
    }


}

File: src/main/kotlin/com/github/sarhatabaot/kraken/core/ticks/TickUtil.kt
package com.github.sarhatabaot.kraken.core.ticks

import java.util.concurrent.TimeUnit

/**
 *
 * @author sarhatabaot
 */
object TickUtil {
    fun secondsToTicks(seconds: Long): Long {
        return 20L * seconds
    }

    fun minutesToTicks(minutes: Int): Long {
        return asTicks(minutes, TimeUnit.MINUTES)
    }

    fun hoursToTicks(hours: Int): Long {
        return asTicks(hours, TimeUnit.HOURS)
    }

    fun asTicks(units: Int, timeUnit: TimeUnit): Long {
        return secondsToTicks(timeUnit.toSeconds(units.toLong()))
    }
}